trinity/main.py

```diff
@@ -139,21 +140,29 @@ def main() -> None:
 
     display_launch_logs(chain_config)
 
-    # if console command, run the trinity CLI
-    if args.subcommand == 'attach':
-        run_console(chain_config, not args.vanilla_shell)
-        sys.exit(0)
-
-    # start the listener thread to handle logs produced by other processes in
-    # the local logger.
-    listener.start()
-
     extra_kwargs = {
         'log_queue': log_queue,
         'log_level': log_level,
         'profile': args.profile,
     }
 
+    # Plugins can provide a subcommand with a `func` which does then control
+    # the entire process from here.
+    if hasattr(args, 'func'):
+        args.func(args, chain_config)
+    else:
+        continue_default_boot(args, chain_config, extra_kwargs, listener, logger)
+
+
+def continue_default_boot(args: Namespace,
```

Comment:
The `continue_` prefix is odd to me.  Maybe we could just call this:

- `trinity_main`
- `trinity_boot`
- `trinity_run`
- `trinity_default`